### ISSUE TITLE
Task export to ObsidianMD results in XML output with ID

### ISSUE THREAD

**Issue by beevelop** [akosbalasko/yarle#434]
I'm trying to export a few thousand notes to Obsidian and it's working well for most parts. Unfortunately, the tasks in Evernote do not get exported properly. I have the following setup:

- yarle `v4.15.1` installed via npm (globally)
- Obsidian `Version 1.1.9 (Installer 1.1.9)`
- Evernote `10.50.10-mac-ddl-public (3842) (Editor: v162.3.20111, Service: v1.62.2)`

<details><summary>Config file (config.json)</summary>
<p>

```json
{
   "enexSources": [
      "/Users/[...]/Absolute Inbox.enex"
   ],
   "templateFile": "/Users/[...]/sampleTemplate.tmpl",
   "outputDir": "/Users/[...]/result",
   "isZettelkastenNeeded": false,
   "useZettelIdAsFilename": false,
   "plainTextNotesOnly": false,
   "skipWebClips": false,
   "useHashTags": true,
   "outputFormat": "ObsidianMD",
   "taskOutputFormat": "ObsidianMD",
   "urlEncodeFileNamesAndLinks": false,
   "skipEnexFileNameFromOutputPath": false,
   "monospaceIsCodeBlock": false,
   "keepMDCharactersOfENNotes": false,
   "keepOriginalAmountOfNewlines": false,
   "addExtensionToInternalLinks": true,
   "nestedTags": {
      "separatorInEN": "_",
      "replaceSeparatorWith": "/",
      "replaceSpaceWith": "-"
   },
   "resourcesDir": "_resources",
   "turndownOptions": {
      "headingStyle": "atx"
   },
   "dateFormat": "YYYY-MM-DD",
   "haveEnexLevelResources": true,
   "haveGlobalResources": false,
   "useUniqueUnknownFileNames": false,
   "logseqSettings": {
      "journalNotes": false
   },
   "obsidianSettings": {
      "omitLinkDisplayName": false
   },
   "obsidianTaskTag": "gtd"
}
```

</p>
</details>

I tried to configure `"taskOutputFormat": "ObsidianMD",` to enable the task export. But with a minimal reproducible example (Note with 4 tasks) it results in the following export in Obsidian:
```xml
<YARLE-EN-V10-TASK>3103aeba-8b27-[...]-9ac5-d8a1cbbd4ab9</YARLE-EN-V10-TASK>
```

Did I miss a setting / is there a misconfiguration?

**Comment by akosbalasko** [comment]
Hi @beevelop ! 

That's strange,  there ```<YARLE-EN-V10-TASK>3103aeba-8b27-[...]-9ac5-d8a1cbbd4ab9</YARLE-EN-V10-TASK>``` lines in the content are temporary placeholders of the tasks, these should be replaced by the real tasks converted to md (this is because of the new implementation tasks in Evernote). 

Could you please send me an example that fails in order to reproduce the problem?
here is my email address: akos0215@gmail.com

Thanks a lot!

**Comment by beevelop** [comment]
@akosbalasko That's what I thought too. I sent you an invite for the example via mail.

**Comment by akosbalasko** [comment]
@beevelop Thank you I have it. I check it and get back to you with the problem (or the fix :) ) asap.

**Comment by akosbalasko** [comment]
@beevelop didn't you miss exporting the Tasks into the Enex file in Evernote? 

<img width="608" alt="Screenshot 2023-01-04 at 23 52 21" src="https://user-images.githubusercontent.com/11886731/210665126-9835bed5-3209-4626-bc0e-3f0e062b9d20.png">

If I select tasks to be exported, Yarle converts the tasks well (the dates are invalid for some reason, but at least the tasks are there), those temporary placeholders are left only if the tasks are not exported at all into the Enex file.

**Comment by t3mujin** [comment]
Just adding my two cents here: I was having the same issue when running over an entire notebook ENEX file, regardless of the taskOutputFormat setting, but when running it over an ENEX file of a single note it worked perfectly. My workaround, and because I had few of them, was to export the notes with tasks separately.

**Comment by beevelop** [comment]
@akosbalasko I checked the full enex file in VScode and can verify that the tasks are contained in the export. That's the format contained in the final result:

```xml
  <note>
    <title>title of the note</title>
    <created>20220728T101921Z</created>
    <updated>20220728T101944Z</updated>
    <note-attributes>
      <author>Maik</author>
      <latitude>[...]</latitude>
      <longitude>[...]</longitude>
      <altitude>[...]</altitude>
      <source>mobile.android</source>
    </note-attributes>
    <content>
      <![CDATA[<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE en-note SYSTEM "http://xml.evernote.com/pub/enml2.dtd"><en-note><div style="--en-task-group:true; --en-id:[...];--en-content-hash:[...];-webkit-user-modify:read-only;-moz-user-modify:read-only;user-modify:read-only;border-radius:3px;border:1px solid rgba(182,182,182,0.09);background:rgba(174,174,174,0.09);overflow:hidden;color:#868686"><div style="background:rgba(182,182,182,0.09) no-repeat 6px 6px url(&quot;data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0iIzg2ODY4NiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTE5LjcyNTEgMjBDMjAuMTgxIDIwIDIwLjYwMTEgMTkuNzU2IDIwLjgyOSAxOS4zNkMyMS4wNTcgMTguOTY0IDIxLjA1NyAxOC40NzYgMjAuODI5IDE4LjA4TDEzLjEwMzEgNC42Mzk5OUMxMi44NzYgNC4yNDM5OSAxMi40NTUxIDQgMTIgNEMxMS41NDQ5IDQgMTEuMTI0IDQuMjQzOTkgMTAuODk2MSA0LjYzOTk5TDMuMTcwOTggMTguMDhDMi45NDMwMSAxOC40NzYgMi45NDMwMSAxOC45NjQgMy4xNzA5OCAxOS4zNkMzLjM5ODEgMTkuNzU2IDMuODE4OTMgMjAgNC4yNzQwOCAyMEgxOS43MjUxWk00Ljg2NDg5IDE4LjRMMTIgNS45ODc5OEwxOS4xMzUxIDE4LjRINC44NjQ4OVoiLz4KPHBhdGggZD0iTTEyLjc1IDEwLjU3NTJDMTIuNzUgMTAuMTYxIDEyLjQxNDIgOS44MjUyMyAxMiA5LjgyNTIzQzExLjU4NTggOS44MjUyMyAxMS4yNSAxMC4xNjEgMTEuMjUgMTAuNTc1MlYxMy4yOTc5QzExLjI1IDEzLjcxMjEgMTEuNTg1OCAxNC4wNDc5IDEyIDE0LjA0NzlDMTIuNDE0MiAxNC4wNDc5IDEyLjc1IDEzLjcxMjEgMTIuNzUgMTMuMjk3OVYxMC41NzUyWiIvPgo8cGF0aCBkPSJNMTIuODc1IDE1Ljk1MTFDMTIuODc1IDE2LjQzNDQgMTIuNDgzMiAxNi44MjYxIDEyIDE2LjgyNjFDMTEuNTE2OCAxNi44MjYxIDExLjEyNSAxNi40MzQ0IDExLjEyNSAxNS45NTExQzExLjEyNSAxNS40Njc5IDExLjUxNjggMTUuMDc2MSAxMiAxNS4wNzYxQzEyLjQ4MzIgMTUuMDc2MSAxMi44NzUgMTUuNDY3OSAxMi44NzUgMTUuOTUxMVoiLz4KPC9zdmc+Cg==&quot;);padding:8px 38px;font-weight:600">Content not supported</div><div style="padding:2px 6px;margin: 1em">This block is a placeholder for Tasks, which has been officially released on the newest version of Evernote and is no longer supported on this version. Deleting or moving this block may cause unexpected behavior in newer versions of Evernote.</div></div><div><br /></div></en-note>      ]]>
    </content>
    <task>
      <title>Lasertag [...]</title>
      <created>[...]</created>
      <updated>[...]</updated>
      <taskStatus>open</taskStatus>
      <taskFlag>false</taskFlag>
      <sortWeight>W</sortWeight>
      <noteLevelID>923d4756-[...]</noteLevelID>
      <taskGroupNoteLevelID>3103aeba-[...]</taskGroupNoteLevelID>
      <dueDate>[...]</dueDate>
      <statusUpdated>[...]</statusUpdated>
      <creator>297[...]</creator>
      <lastEditor>297[...]</lastEditor>
    </task>
    <task>
      <title>Kleine Städtetour</title>
      <created>[...]</created>
      <updated>[...]</updated>
      <taskStatus>open</taskStatus>
      <taskFlag>false</taskFlag>
      <sortWeight>U</sortWeight>
      <noteLevelID>49dde967-[...]</noteLevelID>
      <taskGroupNoteLevelID>3103aeba-[...]</taskGroupNoteLevelID>
      <dueDate>[...]</dueDate>
      <statusUpdated>[...]</statusUpdated>
      <creator>297[...]</creator>
      <lastEditor>297[...]</lastEditor>
    </task>
    <task>
      <title>Paintball</title>
      <created>[...]</created>
      <updated>[...]</updated>
      <taskStatus>open</taskStatus>
      <taskFlag>false</taskFlag>
      <sortWeight>Q</sortWeight>
      <noteLevelID>e991deac-[...]</noteLevelID>
      <taskGroupNoteLevelID>3103aeba-[...]</taskGroupNoteLevelID>
      <dueDate>[...]</dueDate>
      <statusUpdated>[...]</statusUpdated>
      <creator>297[...]</creator>
      <lastEditor>297[...]</lastEditor>
    </task>
    <task>
      <title>Tandem mit Johannes</title>
      <created>[...]</created>
      <updated>[...]</updated>
      <taskStatus>open</taskStatus>
      <taskFlag>false</taskFlag>
      <sortWeight>I</sortWeight>
      <noteLevelID>652fb3ec-[...]</noteLevelID>
      <taskGroupNoteLevelID>3103aeba-[...]</taskGroupNoteLevelID>
      <dueDate>[...]</dueDate>
      <statusUpdated>[...]</statusUpdated>
      <creator>297[...]</creator>
      <lastEditor>297[...]</lastEditor>
    </task>
  </note>
```

**Comment by akosbalasko** [comment]
@t3mujin Thank you for your insights, that was the hint that let me reproduce the problem. 
@beevelop it is fixed in the latest release (4.15.2), please check it out. 

Thank you all!
